src/socket/address.cr

```diff
@@ -204,7 +230,20 @@ class Socket
 
     protected def initialize(sockaddr : LibC::SockaddrUn*, size)
       @family = Family::UNIX
-      @path = String.new(sockaddr.value.sun_path.to_unsafe)
+
+      path = sockaddr.value.sun_path
+      if path[0]? == 0
```

Comment:
`path.starts_with?(Char::ZERO)` ?

Reply:
path is not a `String` here, it's a `StaticArray(UInt8)`